monai/data/utils.py

```diff
@@ -346,9 +347,15 @@ def list_data_collate(batch: Sequence):
             ret = {}
             for k in elem:
                 key = k
-                ret[key] = default_collate([d[key] for d in data])
-            return ret
-        return default_collate(data)
+                data_for_batch = [d[key] for d in data]
+                ret[key] = default_collate(data_for_batch)
+                if isinstance(ret[key], MetaTensor) and all(isinstance(d, MetaTensor) for d in data_for_batch):
```

Comment:
May I know what's the `meta` in the collated `ret[key]` if not all the items are `MetaTensor`?

Thanks in advance.

Reply:
I envisage it being the case where all of them or none of them will be `MetaTensor`.